testing/smoke-ide-test/src/smokeIdeTest/groovy/org/gradle/ide/sync/IsolatedProjectsAndroidProjectSyncTest.groovy

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.ide.sync
+
+import org.gradle.ide.sync.fixtures.IsolatedProjectsIdeSyncFixture
+import org.gradle.test.precondition.Requires
+import org.gradle.test.preconditions.UnitTestPreconditions
+
+class IsolatedProjectsAndroidProjectSyncTest extends AbstractIdeSyncTest {
+
+    // https://developer.android.com/build/releases/gradle-plugin
+    private final static String AGP_VERSION = "8.7.0"
```

Comment:
❓  Can we avoid hardcoding this version here and reuse the existing source of AGP versions?

https://github.com/gradle/gradle/blob/83cfdf63cd100fe5be43ac6b15afe683596400e8/testing/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AbstractSmokeTest.groovy#L98

Reply:
But then we'll tied to AGP versions used in Smoke tests. Given that the goal of sync tests is shaking out IP violations ASAP or freeze existing violations, these tests has to be operating with the latest versions of AGP and be updated separetly. Although I don't feel strongly about this

Reply:
>Given that the goal of sync tests is shaking out IP violations ASAP or freeze existing violations, these tests has to be operating with the latest versions of AGP and be updated separetly.

We smoke test against latest versions of AGP and even nightlies. How do you see the version here being updated more frequently?

We have not been updating the IDEA version for the sync test frequently enough to keep up with their releases. I am concerned we'll have the same situation with the Studio.

Reply:
My question rather would be when we are updating AGP version for the smoke tests? Do we do it regularly?

Reply:
I'd say they are updated relatively regularly:
https://github.com/gradle/gradle/commits/master/gradle/dependency-management/agp-versions.properties

Most importantly, it's a single place where they are updated. But also there is already some infrastructure around it that can be leveraged, if we reuse these AGP version declarations

Reply:
Thanks, applied it